community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v2_3/TransactionBoundQueryContext.scala

```diff
@@ -516,7 +516,7 @@ final class TransactionBoundQueryContext(tc: TransactionalContextWrapper, val re
       tc.kernelTransaction.indexRead().nodeExplicitIndexQuery(cursor, name, query)
       new CursorIterator[Node] {
         override protected def fetchNext(): Node = {
-          if (cursor.next()) proxySpi.newNodeProxy(cursor.nodeReference())
+          if (cursor.next() && reads().nodeExists(cursor.nodeReference())) proxySpi.newNodeProxy(cursor.nodeReference())
```

Comment:
Is this implementation really sound? If there are 3 entries in the index and the middle one is deleted, we will only return the first one, while I argue that we should return the first and the last.